### PR TITLE
Handle zero-length PlTermv

### PR DESCRIPTION
**Pull request by kamahen**
There are a couple of TODOs where I wasn't sure of the correct code - please tell me what's correct and I'll modify this PR (or you can make the changes, of course).

### CODE REVIEW

File: SWI-cpp2.h [hunk 1]
```diff
@@ -587,21 +587,25 @@ class PlTermv
...
 
 public:
+  explicit PlTermv()
+    : size_(0),
+      a0_(PlTerm::null) { }
   explicit PlTermv(size_t n)
     : size_(n),
       a0_(PL_new_term_refs(static_cast<int>(n)))
-  { if ( ! a0_ )
+  { if ( size_ && ! a0_ )
```
Comment (kamahen) [inline SWI-cpp2.h:596]: This should be `size_ && a0_ != PlTerm::null` (even though `PlTerm::null` is 0)

File: SWI-cpp2.h [hunk 2]
```diff
@@ -587,21 +587,25 @@ class PlTermv
...
       throw PlFail();
   }
   explicit PlTermv(size_t n, const PlTerm& t0)
     : size_(n),
       a0_(t0.C_)
-  { if ( ! a0_ )
+  { if ( size_ && ! a0_ )
```
Comment (kamahen) [inline SWI-cpp2.h:602]: This should be `size_ && a0_ != PlTerm::null` (even though `PlTerm::null` is 0)

File: pl2cpp2.doc [hunk 3]
```diff
@@ -2097,24 +2095,26 @@ to Prolog or using \ctype{PlFrame}). Use of some intermediate types
 causing more hash-table lookups. This could be fixed, at the price of
 slighly complicating the interface.
 
-\subsection{Notes on exception}
+\subsection{Notes on exceptions}
```
Comment (kamahen) [inline pl2cpp2.doc:2098]: This documentation should go into a separate PR

**Comment by kamahen** [comment]
@JanWielemaker - Please tell me whether the call to `PL_predicate()` in `PL_open_query` should specify `"user"` or instead there should be a call to `PL_pred` with `module_t` as `0`.

I'll then make new PRs (the documentation change should be in another PR).

**Comment by JanWielemaker** [comment]
> Please tell me whether the call to `PL_predicate()` in `PL_open_query` should specify `"user"` or instead there should be a call to `PL_pred` with `module_t` as `0`.

I think this should be using `PL_pred()` as `PL_predicate()` is just a more high level alternative.   If you already have atoms and modules, going through a string and re-create the atom is an unnecessary detour.    The module parameter should normally be NULL, using the default context.    Only if an explicit module is intended one should specify it.

**Comment by kamahen** [comment]
Removing this PR and will submit a fixed one.
